docs/website/docs/reference/tuning.md

```diff
@@ -0,0 +1,126 @@
+
+transform.named_sequence
+@match_mmt_2048x1280x5120_f16_f16_f32(%matmul: !transform.any_op {transform.readonly})
+  -> (!transform.any_op, !transform.any_param) {
+  %mmt = transform.include @match_mmt_f16_f16_f32 failures(propagate) (%matmul)
+    : (!transform.any_op) -> !transform.any_op
+  %lhs = transform.get_operand %matmul[0] : (!transform.any_op) -> !transform.any_value
+  %rhs = transform.get_operand %matmul[1] : (!transform.any_op) -> !transform.any_value
+  transform.iree.match.cast_compatible_type %lhs = tensor<2048x5120xf16> : !transform.any_value
+  transform.iree.match.cast_compatible_type %rhs = tensor<1280x5120xf16> : !transform.any_value
+  %config = transform.param.constant #iree_codegen.compilation_info<
+    lowering_config = #iree_gpu.lowering_config<{promote_operands = [0, 1],
+                                                 mma_kind = #iree_gpu.mma_layout<MFMA_F32_16x16x16_F16>,
+                                                 subgroup_m_count = 2, subgroup_n_count = 2,
+                                                 reduction = [0, 0, 64],
+                                                 workgroup = [64, 128, 0]}>,
+    translation_info = #iree_codegen.translation_info<pipeline = LLVMGPUVectorDistribute
+      workgroup_size = [256, 1, 1] subgroup_size = 64,
+      {gpu_pipeline_options = #iree_gpu.pipeline_options<prefetch_shared_memory = true>}>
+  > -> !transform.any_param
+  transform.yield %matmul, %config : !transform.any_op, !transform.any_param
+}
+
+transform.named_sequence
+@__kernel_config(%variant_op: !transform.any_op {transform.consumed}) -> !transform.any_op
+  attributes { iree_codegen.tuning_spec_entrypoint } {
+  %res = transform.foreach_match in %variant_op
+    @match_mmt_2048x1280x5120_f16_f16_f32 -> @apply_op_config
+    : (!transform.any_op) -> !transform.any_op
+  transform.yield %res : !transform.any_op
+}
+}
+```
+
+### Explanation
+
+Tuning specs are
+[transform dialect](https://mlir.llvm.org/docs/Dialects/Transform/) libraries
+that conform to the following format:
+
+* All tuning spec entry points (named sequence ops) are marked with the
+  `iree_codegen.tuning_spec_entrypoint` attribute. They have a single argument
+  of type `!transform.any_op` and return a single value of type
+  `!transform.any_op`.
+* All entry points in the final tuning specs must either read
+  (`transform.readonly`) or consume (`transform.consumed`) the argument.
+
+The tuning spec above attempts to match `linalg.generic` ops that correspond to the
+matmul operation with the RHS operand transposed (a.k.a. mmt) of shape
+`2048x1280x5120` and `f16` operand element types and `f32` result element type.
+
+If the match succeeds, the tuning spec applies the `compilation_info` attribute
+that will drive the code generation. This attribute is considered a compiler
+implementation detail; in general, each codegen pipeline has its own
+requirements wrt what is considered a valid compilation info and how to
+interpret it.
```

Comment:
Prefer to spell out "with respect to" instead of abbreviating, or use other language.

https://developers.google.com/style/abbreviations#dont-use

> Don't use internet slang abbreviations such as [tl;dr](https://developers.google.com/style/word-list#tldr), [ymmv](https://developers.google.com/style/word-list#ymmv), [RTFM](https://developers.google.com/style/word-list#rtfm), or others. Write out what you mean in a non-figurative way.
> Use the most common form of a word. If the spelled-out word is common and easily understandable, use that rather than abbreviating. For example, write approximately instead of approx.